tagstudio/src/core/library/alchemy/db.py

```diff
@@ -0,0 +1,48 @@
+from pathlib import Path, PurePath
+
+import structlog
+from sqlalchemy import Dialect, Engine, String, TypeDecorator, create_engine, text
+from sqlalchemy.orm import DeclarativeBase
+
+logger = structlog.getLogger(__name__)
```

Comment:
for now I just want to focus on the actual database rather than niceties around. Using default `logging` is cumbersome so I replaced it with `structlog`, but I dont want to think where and how to have the logging placed and configured, so I use this approach for now. This can sure be refactored out later.


> how would structlog handle threading

Since it shows the logs from other threads by default, I dont really dig deep into it


> And for windows you also need colorama for colors; and Rich or better-exceptions if you want more helpful output.

I'm happy with the default exceptions, as that will be better when eventually someone will be copy-pasting the traceback when reporting a bug.